OpenEdXMobile/src/main/java/org/edx/mobile/view/MyCoursesListFragment.kt

```diff
@@ -0,0 +1,256 @@
+            environment?.database?.updateAllVideosAsDeactivated(dataCallback)
+            for (i in list.indices) { //Check if the flag of isIs_active is marked to true,
+                //then activate all videos
+                if (list[i].isIs_active) { //update all videos for a course fetched in the API as Activated
+                    environment?.database?.updateVideosActivatedForCourse(list[i].course.id,
+                            dataCallback)
+                } else {
+                    list.removeAt(i)
+                }
+            }
+            //Delete all videos which are marked as Deactivated in the database
+            environment?.storage?.deleteAllUnenrolledVideos()
+        }
+    }
+
+    private val dataCallback: DataCallback<Int> = object : DataCallback<Int>() {
+        override fun onResult(result: Int) {}
+        override fun onFail(ex: Exception) {
+            logger.error(ex)
+        }
+    }
+
+    override fun onResume() {
+        super.onResume()
+        if (refreshOnResume) {
+            loadData(showProgress = false, isCached = true)
+            refreshOnResume = false
+        }
+    }
+
+    override fun onDestroy() {
+        super.onDestroy()
+        EventBus.getDefault().unregister(this)
+    }
+
+    fun onEventMainThread(event: EnrolledInCourseEvent?) {
+        refreshOnResume = true
+    }
+
+    private fun loadData(showProgress: Boolean, isCached: Boolean) {
+        if (showProgress) {
+            binding.loadingIndicator.root.visibility = View.VISIBLE
+            errorNotification.hideError()
+        }
+        if (isCached) {
+            getUserEnrolledCourses(api?.enrolledCoursesFromCache!!)
+        }
+        getUserEnrolledCourses(api?.enrolledCourses!!)
+    }
+
+    private fun invalidateView() {
+        binding.swipeContainer.isRefreshing = false
+        binding.loadingIndicator.root.visibility = View.GONE
+
+        if (!EventBus.getDefault().isRegistered(this@MyCoursesListFragment)) {
+            EventBus.getDefault().registerSticky(this@MyCoursesListFragment)
+        }
+    }
+
+    private fun showError(error: Throwable) {
+        context?.let { context ->
+            error.cause?.let { cause ->
+                errorNotification.showError(context, cause, R.string.lbl_reload) {
+                    if (NetworkUtil.isConnected(context)) {
+                        onRefresh()
+                    }
+                }
+            }
+        }
+    }
+
+    private fun getUserEnrolledCourses(enrolledCoursesApi: Call<List<EnrolledCoursesResponse>>) {
+        enrolledCoursesApi.enqueue(object : Callback<List<EnrolledCoursesResponse>> {
+            override fun onResponse(call: Call<List<EnrolledCoursesResponse>>, response: Response<List<EnrolledCoursesResponse>>) {
+                if (response.isSuccessful && response.code() == HttpStatus.OK) {
+                    populateCourseData(ArrayList(response.body()))
```

Comment:
I think we have to add [L54-L55](https://github.com/edx/edx-app-android/pull/1438/files#diff-4bbef73d3265ad625b76074453f9e90fL54-L55) may be open edx community uses those lines of code?

@miankhalid thoughts?

Reply:
not need to add.